homeassistant/components/sma/config_flow.py

```diff
@@ -0,0 +1,196 @@
+            self._data[CONF_SSL] = user_input[CONF_SSL]
+            self._data[CONF_VERIFY_SSL] = user_input[CONF_VERIFY_SSL]
+            self._data[CONF_GROUP] = user_input[CONF_GROUP]
+            self._data[CONF_PASSWORD] = user_input[CONF_PASSWORD]
+
+            try:
+                self._data[DEVICE_INFO] = await validate_input(self.hass, user_input)
+            except aiohttp.ClientError:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except CannotRetrieveDeviceInfo:
+                errors["base"] = "cannot_retrieve_device_info"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if not errors:
+                await self.async_set_unique_id(self._data[DEVICE_INFO]["serial"])
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=self._data[CONF_HOST], data=self._data
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=self._data[CONF_HOST]): cv.string,
+                    vol.Optional(CONF_SSL, default=self._data[CONF_SSL]): cv.boolean,
+                    vol.Optional(
+                        CONF_VERIFY_SSL, default=self._data[CONF_VERIFY_SSL]
+                    ): cv.boolean,
+                    vol.Optional(CONF_GROUP, default=self._data[CONF_GROUP]): vol.In(
+                        GROUPS
+                    ),
+                    vol.Required(CONF_PASSWORD): cv.string,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_config=None):
+        """Import a config flow from configuration."""
+        device_info = await validate_input(self.hass, import_config)
+        config_entry_unique_id = device_info["serial"]
+        import_config[DEVICE_INFO] = device_info
+
+        config_sensors = import_config[CONF_SENSORS].copy()
+
+        # find and replace sensors removed from pysma
+        for sensor in config_sensors:
+            if sensor in pysma.LEGACY_MAP:
+                import_config[CONF_SENSORS].remove(sensor)
+                import_config[CONF_SENSORS].append(
+                    pysma.LEGACY_MAP[sensor]["new_sensor"]
+                )
+
+        # If unique is configured import was already run
+        # This means remap was already done, so we can abort
+        await self.async_set_unique_id(config_entry_unique_id)
+        self._abort_if_unique_id_configured(import_config)
+
+        entity_registry = er.async_get(self.hass)
+
+        # Init all default sensors
+        sensor_def = pysma.Sensors()
+
+        # Add sensors from the custom config
```

Comment:
This code is duplicated in `__init__.py` line 47.  Can you break this out into a shared function to make it more DRY?

Reply:
Moved to `__init__.py` in a19b5a1